shim/internal/config.go

```diff
@@ -87,20 +87,63 @@ func LoadConfig() (Config, error) {
 		return Config{}, fmt.Errorf("failed to read root cert file: %s", err)
 	}
 
-	rootCertPool := x509.NewCertPool()
-	if ok := rootCertPool.AppendCertsFromPEM(root); !ok {
-		return Config{}, errors.New("failed to load root cert file")
+	tlscfg, err := LoadTLSConfig(false, key, cert, root)
+	if err != nil {
+		return Config{}, err
+	}
+
+	conf.TLS = tlscfg
+
+	return conf, nil
+}
+
+// LoadTLSConfig ...
```

Comment:
Another comment better filled in or deleted.

Reply:
will delete.